server/src/main/java/org/elasticsearch/common/xcontent/XContentBuilder.java

```diff
@@ -634,63 +632,32 @@ public XContentBuilder utf8Value(byte[] bytes, int offset, int length) throws IO
     // Date
     //////////////////////////////////
 
-    public XContentBuilder field(String name, ReadableInstant value) throws IOException {
-        return field(name).value(value);
-    }
-
-    public XContentBuilder field(String name, ReadableInstant value, DateTimeFormatter formatter) throws IOException {
-        return field(name).value(value, formatter);
-    }
-
-    public XContentBuilder value(ReadableInstant value) throws IOException {
-        return value(value, DEFAULT_DATE_PRINTER);
-    }
-
-    public XContentBuilder value(ReadableInstant value, DateTimeFormatter formatter) throws IOException {
-        if (value == null) {
-            return nullValue();
-        }
-        ensureFormatterNotNull(formatter);
-        return value(formatter.print(value));
-    }
-
-    public XContentBuilder field(String name, Date value) throws IOException {
-        return field(name).value(value);
-    }
-
-    public XContentBuilder field(String name, Date value, DateTimeFormatter formatter) throws IOException {
-        return field(name).value(value, formatter);
-    }
-
-    public XContentBuilder value(Date value) throws IOException {
-        return value(value, DEFAULT_DATE_PRINTER);
+    public XContentBuilder timeField(String name, Object timeValue) throws IOException {
+        return field(name).timeValue(timeValue);
     }
 
-    public XContentBuilder value(Date value, DateTimeFormatter formatter) throws IOException {
-        if (value == null) {
-            return nullValue();
-        }
-        return value(formatter, value.getTime());
-    }
-
-    public XContentBuilder dateField(String name, String readableName, long value) throws IOException {
+    public XContentBuilder timeField(String name, String readableName, long value) throws IOException {
```

Comment:
Why is this version needed? Can't callers construct a Date from the long value? I think at least these methods need javadocs.

Reply:
This version is needed because these fields work differently than the other human readable fields, where the coercion takes place only if human readable is needed, otherwise the raw numeric value is used. I'll add javadocs to the methods